pkg/deploy/exec_commands.go

```diff
@@ -72,7 +72,7 @@ func GetKeycloakProvisionCommand(cr *orgv1.CheCluster, cheHost string) (command
 	createRealmClientUserCommand := r.Replace(str)
 	command = createRealmClientUserCommand
 	if cheFlavor == "che" {
-		command = "cd /scripts && " +createRealmClientUserCommand
+		command = "cd /scripts && export JAVA_TOOL_OPTIONS=-Duser.home=. && " +createRealmClientUserCommand
```

Comment:
Why do you need to set user.home=. ?

Reply:
Let me explain a bit more:
- The Keycloak docker image for the `che` (=> upstream) case is [`eclipse/che-keycloak`](https://github.com/eclipse/che/blob/master/dockerfiles/keycloak/Dockerfile). And as you can see from the source code, it uses user `root`. So on Openshift, the used user becomes an arbitrary user. So finally the only place where the user has write access is the [`scripts` directory](https://github.com/eclipse/che/blob/master/dockerfiles/keycloak/Dockerfile#L16).
- The operator code tries to run the `kcadm.sh` CLI in the POD `keycloak` container through exec. But the problem is that by default the `kcadm` login process tries to create a file in: `$HOME/.keycloak/kcadm.config`... well more precisely in `<java user.home system property>//.keycloak/kcadm.config` since it's a Java process.
- This is why I add the reset of the `user.home` before running the various `kcadm` commands.

Reply:
@davidfestal The user should **not** be `root` but the user with `UID 1000`. At least that's how the base image [has been designed](https://github.com/jboss-dockerfiles/keycloak/blob/master/server/Dockerfile#L24). We are changing the user in our [Dockerfile](https://github.com/eclipse/che/blob/master/dockerfiles/keycloak/Dockerfile) to be able to `chown` and `chgrp` but we should revert it to user `1000` afterwards. I may miss something but it looks like an issue in our Dockerfile.

Reply:
Yes, I think the same way as well, and would probably open a PR on the `che-keycloak` docker image in the future. But in order to avoid any unmanaged side-effect of coming back to the `1000` user ID too quickly, I proposed this change as a temporary solution. Does it seem an acceptable temporary solution for you ?

Reply:
@davidfestal I am ok with doing that in a separate PR